### ISSUE TITLE
make build verbose

### ISSUE THREAD

**Issue by maxbachmann** [scikit-build/scikit-build#752]
when building as verbose using `pip install -v` it would be helpful, if scikit-build would set the `VERBOSE` environment variable. This would help in debugging broken builds, by printing e.g. the compiler arguments.

**Comment by henryiii** [comment]
I don't think we can access the number of "-v"'s, and I wouldn't want to go from nothing to showing _both_ CMake issues and VERBOSE=1 compiler commands with just one `-v`. Scikit-build-core will have a configuration system & you'd be able to set verbose=true in the TOML file or by PEP 517 config options.

**Comment by henryiii** [comment]
(If we _can_ detect the number of `-v`'s passed, this would be fantastic, though, I think.

**Comment by henryiii** [comment]
scikit-build (classic) after #1185 is now a thin wrapper around scikit-build-core's setuptools plugin. Closing all open issues; if this still applies, please reopen or open an issue on scikit-build-core. Thanks!
